binderhub/static/js/index.js

```diff
@@ -231,19 +228,35 @@ function build(providerSpec, log, path, pathType) {
 }
 
 function setUpLog() {
-  var log = new Terminal({
+  const log = new Terminal({
     convertEol: true,
     disableStdin: true
   });
+  const fitAddon = new FitAddon();
+  log.loadAddon(fitAddon);
+
+  var $logContainer = $("div#log-container");
+  var $panelBody = $("div.panel-body");
+  log.fit = function () {
+    if (!$panelBody.hasClass('hidden')) {
```

Comment:
To prevent errors, fit is called only if parent is visible. Because fit is called every time window is resized (see [line 256](https://github.com/jupyterhub/binderhub/pull/1248/files#diff-ecd046a09fce6b476b2a7e29adaae5537510474c0431c720b4a973d38e51b595R256-R258) and this may happen when panel is hidden.